### PR TITLE
Update accounts.cs-e2e.ts

### PR DESCRIPTION
**Pull request by cfaur09**
## Proposed Changes
- Update accounts.cs-e2e.ts

## How to test
- npm run test:cs-e2e accounts.cs-e2e.ts

### CODE REVIEW

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 1]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?sender=${config.aliceAddress}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
+    });
+
+    it('should return transfers with receiver parameter', async () => {
+      const receiverAddress = 'erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u';
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?receiver=${receiverAddress}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
Comment (bogdan-rosianu) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1214]: check the receiver of the results

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 2]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      const token = accountTokens.data[0].identifier;
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?token=${token}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
+    });
+
+    it('should return transfers with senderShard parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?senderShard=1`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
Comment (bogdan-rosianu) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1228]: check the sender shard of the results

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 3]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with receiverShard parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?receiverShard=4294967295`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
Comment (bogdan-rosianu) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1234]: check the receiver shard of the results

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 4]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?miniBlockHash=${miniBlockHash}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(1);
+    });
+
+    it('should return transfers with hashes parameter', async () => {
+      const hashes = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=2`);
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?hashes=${hashes.data[0].txHash},${hashes.data[1].txHash}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(2);
```
Comment (bogdan-rosianu) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1249]: check the hashes of the results

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 5]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    it('should support pagination', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=2`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(2);
+    });
+
+    it('should return transfers with sender parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?sender=${config.aliceAddress}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1207]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 1 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1218]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 6]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with token parameter', async () => {
+      const accountTokens = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/tokens`);
+      const token = accountTokens.data[0].identifier;
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?token=${token}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1230]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 2 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1240]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 3 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1250]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 7]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with miniBlockHash parameter', async () => {
+      const miniBlockHashes = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=1`);
+      const miniBlockHash = miniBlockHashes.data[0].miniBlockHash;
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?miniBlockHash=${miniBlockHash}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1262]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 4 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1273]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 8]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      const hashes = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=2`);
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?hashes=${hashes.data[0].txHash},${hashes.data[1].txHash}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(2);
+    });
+
+    it('should return transfers with status parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?status=success`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1281]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 9]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with round parameter', async () => {
+      const rounds = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=1`);
+      const round = rounds.data[0].round;
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?round=${round}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1293]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 10]
```diff
@@ -1187,4 +1187,232 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with senderOrReceiver parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?senderOrReceiver=${config.aliceAddress}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1303]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 11]
```diff
@@ -1261,12 +1291,20 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      }
     });
 
     it('should return transfers with senderOrReceiver parameter', async () => {
       const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?senderOrReceiver=${config.aliceAddress}`);
       expect(response.status).toBe(200);
       expect(response.data.length).toBeGreaterThanOrEqual(1);
+
+      for (const transfer of response.data) {
+        expect(transfer.receiver).toBe('erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u');
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1306]: yes, I will add now the new expect

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1230]: it is not done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 12]
```diff
@@ -1187,4 +1187,270 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      }
+    });
+
+    it('should return transfers with withLogs parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?withLogs=true`);
+      expect(response.status).toBe(200);
+      const hasLogs = response.data.some((transfer: any) => transfer.logs);
+      expect(hasLogs).toBe(true);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1314]: i would check also at least one transfer to have specific (expected) logs

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 13]
```diff
@@ -1187,4 +1187,270 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with withOperations parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?withOperations=true`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
+      const hasOperations = response.data.some((transfer: any) => transfer.operations && transfer.operations.length > 0);
+      expect(hasOperations).toBe(true);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1322]: i would check also at least one transfer to have specific (expected) logs

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 14]
```diff
@@ -1187,4 +1187,270 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(1);
+
+      for (const transfer of response.data) {
+        expect(transfer.miniBlockHash).toBe(miniBlockHash);
+      }
+    });
+
+    it('should return transfers with hashes parameter', async () => {
+      const hashes = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=2`);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1270]: create a global variable and use it.
At the beginning of the new test, you can define a local variable endpoint = "/.."

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1306]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 14 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1282]: no needed

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 13 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1345]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 12 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1326]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 15]
```diff
@@ -1187,4 +1187,270 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    it('should support pagination', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?size=2`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toStrictEqual(2);
+    });
+
+    it('should return transfers with sender parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?sender=${config.aliceAddress}`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 16]
```diff
@@ -1187,4 +1187,270 @@ describe('Accounts e2e tests with chain simulator', () => {
       expect(response.data).toBeGreaterThanOrEqual(1);
     });
   });
+
+  describe('GET /accounts/:address/transfers', () => {
+    it('should return transfers for a given address', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers`);
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
```
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts]: done

**Review by bogdan-rosianu — DISMISSED** [review]

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 17]
```diff
@@ -1187,4 +1191,346 @@ describe('Accounts e2e tests with chain simulator', () => {
       expect(response.data).toBeGreaterThanOrEqual(1);
     });
   });
+
+  describe('GET /accounts/:address/transfers', () => {
+    it('should return transfers for a given address', async () => {
+      const countResponse = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers/count`);
+      const expectedCount = countResponse.data;
+
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers`);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1200]: have a check that also the count si > 0. otherwise both endpoints may return no data/count 0

Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1202]: done

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 18]
```diff
@@ -1187,4 +1191,346 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+    });
+
+    it('should return transfers with token parameter', async () => {
+      const accountTokens = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/tokens`);
+      const token = accountTokens.data[0].identifier;
+
+      const countResponse = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers/count?token=${token}`);
+      const expectedCount = countResponse.data;
+
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?token=${token}`);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1244]:
```suggestion
      const endpoint = `{config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers`
      const countResponse = await axios.get(`${endpoint}/count?token=${token}`);
      const expectedCount = countResponse.data;

      const response = await axios.get(`${endpoint}?token=${token}`);
```

easier to follow and less errors

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 19]
```diff
@@ -1187,4 +1191,346 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      expect(response.data.length).toStrictEqual(expectedCount);
+
+      for (const transfer of response.data) {
+        expect(transfer.action.arguments.transfers[0].token).toBe(token);
+        expect(transfer.function).toBe('ESDTTransfer');
+      }
+    });
+
+    it('should return transfers with senderShard parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?senderShard=1`);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1255]: this test can be combined with `should return the total number of transfers for a given address with senderShard parameter`

File: src/test/chain-simulator/accounts.cs-e2e.ts [hunk 20]
```diff
@@ -1187,4 +1191,348 @@ describe('Accounts e2e tests with chain simulator', () => {
...
+      expect(response.status).toBe(200);
+      expect(response.data.length).toBeGreaterThanOrEqual(1);
+
+      for (const transfer of response.data) {
+        expect(transfer.senderShard).toBe(1);
+      }
+    });
+
+    it('should return transfers with receiverShard parameter', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/accounts/${config.aliceAddress}/transfers?receiverShard=4294967295`);
```
Comment (dragos-rebegea) [inline src/test/chain-simulator/accounts.cs-e2e.ts:1265]: this test can be combined with `should return the total number of transfers for a given address with receiverShard parameter`

Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1265]: No, I will leave the test as it is

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 19 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1255]: No, I will leave the test as it is

File: src/test/chain-simulator/accounts.cs-e2e.ts (hunk 18 above)
Comment (cfaur09) [reply src/test/chain-simulator/accounts.cs-e2e.ts:1244]: what errors?
it's clear that we have `countResponse` that returns a number ( `count` ) and that assign the value to the expectedCount value.

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**